internal/controllers/extension_controller.go

```diff
@@ -116,29 +137,101 @@ func (r *ExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alpha1.Ext
 		return ctrl.Result{}, nil
 	}
 
+	if !r.HasKappApis {
+		setInstalledStatusConditionFailed(&ext.Status.Conditions, errkappAPIUnavailable.Error(), ext.GetGeneration())
+
+		ext.Status.ResolvedBundleResource = ""
+		setResolvedStatusConditionUnknown(&ext.Status.Conditions, "kapp apis are unavailable", ext.GetGeneration())
+
+		setDeprecationStatusesUnknown(&ext.Status.Conditions, "kapp apis are unavailable", ext.GetGeneration())
+		return ctrl.Result{}, errkappAPIUnavailable
+	}
+
 	// Don't do anything if Paused
 	ext.Status.Paused = ext.Spec.Paused
 	if ext.Spec.Paused {
 		l.Info("resource is paused", "name", ext.GetName(), "namespace", ext.GetNamespace())
 		return ctrl.Result{}, nil
 	}
 
-	// TODO: kapp-controller integration
-	// * gather variables for resolution
-	// * perform resolution
-	// * lookup the bundle in the solution/selection that corresponds to the Extension's desired Source
-	// * set the status of the Extension based on the respective deployed application status conditions.
+	// TODO: Instead of using a SAT solver and generating variables based on installed and
+	// available bundles, work on a filtering mechanism to evaluate if the bundle and its
+	// APIs are installed on a cluster or not. Feature tracked in a separate issue.
```

Comment:
We don't need to evaluate if the bundle and its APIs are already installed on the cluster.

For now, the "resolution" without deppy would be:
1. Get package, channel, bundle info from catalogd for `ext.Spec.Source.Package.Name`
2. Set `candidates` to all the bundles in the package.
3. If `ext.Spec.Source.Package.Channel` is set, remove candidates unless they have an `entries[].name` in the named channel.
4. If `ext.Spec.Source.Package.Version` is set, remove candidates unless their version is in the specified range.
5. If an `App` already exists for the Extension, remove candidates that are not the existing installed bundle or a successor of the existing installed bundle. (Ideally this would also account for UpgradeConstraintPolicy and the ForceSemver feature gate).
6.  If no candidates remain, "resolution" fails
7. Sort the remaining candidates by semver, and pick the highest.

So this is all just "grab from the catalog(s) and filter based on what's in the spec and what is installed"

Feel free to capture in a follow-up!

Reply:
I've captured this for a follow up #654 and done basic filtering based on version, package and channel. Meanwhile, looking into the code, realized that we do have helpers in place to filter out catalog bundles based on relevant info from extension that currently deppy uses to create variables. We can re-use them to make filtering cleaner.